gtest/src/mailbox.rs

```diff
@@ -0,0 +1,320 @@
+use crate::{manager::ExtManager, CoreLog, Log, RunResult};
+use codec::Encode;
+use gear_core::{
+    ids::{MessageId, ProgramId},
+    message::{Dispatch, DispatchKind, Message, StoredMessage},
+};
+use std::cell::RefCell;
+
+pub struct Mailbox<'a> {
+    manager: &'a RefCell<ExtManager>,
+    user_id: ProgramId,
+}
+
+impl<'a> Mailbox<'a> {
+    pub(crate) fn new(user_id: ProgramId, manager: &'a RefCell<ExtManager>) -> Mailbox<'a> {
+        Mailbox { user_id, manager }
+    }
+
+    pub fn contains<T: Into<Log> + Clone>(&self, log: &T) -> bool {
+        let log: Log = log.clone().into();
+        if let Some(mailbox) = self.manager.borrow().mailbox.get(&self.user_id) {
+            return mailbox.iter().any(|message| log.eq(message));
+        }
+        self.manager
+            .borrow_mut()
+            .mailbox
+            .insert(self.user_id, Vec::default());
+        false
+    }
+
+    pub fn take_message<T: Into<Log>>(&self, log: T) -> MessageReplier {
+        let log: Log = log.into();
+        let mut manager = self.manager.borrow_mut();
+        let index = if let Some(mailbox) = manager.mailbox.get(&self.user_id) {
+            mailbox
+                .iter()
+                .position(|message| log.eq(message))
+                .expect("No message that satisfies log")
+        } else {
+            panic!("No mailbox associated with this program id");
```

Comment:
```suggestion
            panic!("Infallible. No mailbox associated with this user id");
```

Reply:
Are you sure about infallible? I have a test that is catching this panic :)
If we create a mailbox and then in ```ExtManeger``` in ```mailbox``` delete user associated with it "by hands"  we will catch this panic

Reply:
> Are you sure about infallible? I have a test that is catching this panic :)
> If we create a mailbox and then in `ExtManeger` in `mailbox` delete user associated with it "by hands" we will catch this panic

End-user of the lib isn't able to interact with ext manager